mailbox/api/src/main/java/org/apache/james/mailbox/model/UidValidity.java

```diff
@@ -0,0 +1,107 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
+
+import java.security.SecureRandom;
+import java.util.function.Supplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+
+public class UidValidity {
+    private static final SecureRandom RANDOM = new SecureRandom();
+    private static final long UPPER_EXCLUSIVE_BOUND = 4294967296L;
+
+    /**
+     * Despite RFC-3501 recommendations, we chose random as a UidVality generation mechanism.
+     * Timestamp base approach can lead to UidValidity being the same for two mailboxes simultaneously generated, leading
+     * to some IMPA clients not being able to detect changes.
+     *
+     * See https://issues.apache.org/jira/browse/JAMES-3074 for details
+     */
+    public static  UidValidity random() {
+        return fromSupplier(RANDOM::nextLong);
+    }
+
+    @VisibleForTesting
+    static UidValidity fromSupplier(Supplier<Long> longSupplier) {
+        long randomValue = Math.abs(longSupplier.get());
+        long sanitizedRandomValue = 1 + (randomValue % (UPPER_EXCLUSIVE_BOUND - 1));
+        return ofValid(sanitizedRandomValue);
+    }
+
+    public static UidValidity of(long uidValidity) {
```

Comment:
Can you add the comment about the usage of this method only for deserialization ?